### DISCUSSION TITLE
Fallback order or playlist contidional

### DISCUSSION THREAD

**Discussion by sauramud** [savonet/liquidsoap#2428] [Q&A]
Hello everyone, I've tried a lot of configuration but I can't get them works. I was looking for doing a conditional playlist, for example i define this 

`xmaslist = interactive.bool("xmas_playlist", false)`

ok, with this I can use var.set to change from true or false this value.
Then I've got two main playlist, daymix and xmasmix and obviusly a q structure for the requests; in my switch 

```
dc =    switch([
  (predicate.once({10h30m-10h45m}), jingle),
  ({ true }, fallback(track_sensitive=false,[
    ({true} , request.queue(id="q")),
    (xmaslist , xmasmix ),daymix
  ]))
```
Is this correct or not ? I want that: when a request is pulled this is immediatly handled, then, when xmaslist bool is set to true then that playlist works, eventually it was false then the default daymix should be played.

and another question, If I want that the jingle playlist from above is played immediatly at 10h30m how can I archive this ? when I put the track_sensitive out from the fallback I get a syntax error.

Any help will be apreciated, thanks a lot!

**Comment by toots** [comment]
I'm not sure why you want to do this nesting, could this work:
```ruby
dc = switch([
  (predicate.once({10h30m-10h45m}), jingle),
  ({true} , request.queue(id="q")),
  (xmaslist , xmasmix),
  ({true}, daymix)
])
```

**Comment by sauramud** [comment]
Hi @toots 
Yes, that was the correct solution (in these days I've tried a lot of tests and the script you wrote to me is correct, but I go forward with the configuration of a more complex script.
In all my script (modified a lot) I want that some playlists works on the exact timing (then I've got to use the `track_sensitive=false`) and in other must be true while in other one the track_sensitive is set to true so that the song is correctly terminated. Is this a possible configuration? In this moment my fallback / switch is this but something is not working well

```
dc =   fallback(track_sensitive=true,[request.queue(id="q"),
                       switch(track_sensitive=false,[
                               ({(7h30-7h34) and (1w-5w)}   ,delay(1800.,  opens)),
                               ({10h30-10h35}               ,delay(1800.,  chuck)),
                               ({(12h00-12h04) and (1w-4w)} ,delay(1800., lunch)),
                               ({(12h30-12h34) and 5w}      ,delay(1800., lunch)),
                       ]),
                       switch(track_sensitive=true,[
                               ({(11h30-12h30) and 5w }                 ,  latin),
                               ({13h30-14h30}               ,delay(2400.,    mix)),
                               ({ 22h-7h }                              ,  anni80),
                               (xmaslist, xmasmix),
                               ({true}, daymix)
                       ])
               ])
```

And in the `switch/track_sensitive=false` part it seem that the predicate.once is not working correctly and is ignored so I've got to use the delay. Have you ever tried this one ?
And just one last question, when `track_sensitive=false` the song is immediatly interrupted for the jingle / song and after this last one is finisched the previous song restart exactly from where it was interrubped. Is there an option to skip the remaining part of the song and start with a new one after the jingle in the `track_sensitive=false` part ?
Thanks a lot for support :-)
Luke

**Comment by toots** [comment]
I just pushed a cleanup of the request pool implementation and it looks like it did a lot of good to the memory and CPU usage:
![cpu (1)](https://user-images.githubusercontent.com/871060/174463665-186ba1b8-d418-43cf-8822-d5b2de9010c0.png)

I'd suggest to test the latest rolling release for https://github.com/savonet/liquidsoap/commit/5907040dc4b7681abce427f9154b0fa1afc16842
